ocaml/lexer.ml

```diff
@@ -30,6 +34,12 @@ type lexer = {
 let slurp fname = In_channel.with_file ~binary:false fname 
     ~f:(fun ch -> In_channel.input_all ch)
 
+let op_info op = 
+    match op with
+    | Pow -> (3, Right)
+    | Plus | Minus -> (1, Left)
+    | Times | Divide -> (2, Left)
+
```

Comment:
Can't comment on your lexer implementation below since it's not part of this PR, but it looks good. Only comment is I think everywhere you use `!` you don't need the extra parentheses.

Reply:
I get compilation errors when this happens. I suspect it's because ! takes precedence over ., so it tries to get the `.pos` field of a lexer ref.

Reply:
That's very sad. No reason for precedence to be that way.